com/williamfiset/datastructures/binarysearchtree/SplayTree.java

```diff
@@ -0,0 +1,274 @@
+import com.williamfiset.datastructures.utils.TreePrinter;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
+class BinaryTree<T extends Comparable<T>> implements TreePrinter.PrintableNode {
+    private BinaryTree<T> left, right;
+    private T data;
+
+    public BinaryTree(T data) {
+        this.data = data;
+    }
+
+    @Override
+    public BinaryTree<T> getLeft() {
+        return left;
+    }
+
+    @Override
+    public BinaryTree<T> getRight() {
+        return right;
+    }
+
+    @Override
+    public String getText() {
+        return data.toString();
+    }
+    public void setRight(BinaryTree<T> right) {
+        this.right = right;
+        //   right.parent = this;
+    }
+    public void setLeft(BinaryTree<T> left) {
+        this.left = left;
+        // left.parent = this;
+    }
+    public T getData() {
+        return data;
+    }
+
+    public void setData(T data) {
+        this.data = data;
+    }
+
+    @Override
+    public String toString() {
+
+        return TreePrinter.getTreeDisplay(this);
+    }
+
+
+}
+
+/**Note : Each Operation of a SplayTree tries returns the new splayed root after the operation **/
+public class SplayTree<T extends Comparable<T>> {
+
+    private BinaryTree<T> root;
+
+    public SplayTree() {
+        this.root = null;
+    }
+
+    public SplayTree(BinaryTree<T> root) {
+        this.root = root;
+    }
+
+    public BinaryTree<T> getRoot() {
+        return root;
+    }
+
+    private BinaryTree<T> rightRotate(BinaryTree<T> x) {
```

Comment:
Can you give these variables more meaningful names? E.g 'node', 'leftChild', and etc

Reply:
Oke, sure